.github/workflows/lrc-qa.yaml

```diff
@@ -1,34 +1,99 @@
 name: Validate debian/copyright
 
 env:
-  apt_deps: "licenserecon"
-
-# on:
-#   push:
-#     branches:
-#       - main
-#     tags:
-#       - "*"
-#   pull_request:
-#     paths:
-#       - "debian/copyright"
-#       - "debian/lrc.config"
-#       - "go.mod"
-on: [workflow_dispatch]
+  lrc_build_deps: "build-essential debhelper-compat fp-compiler fp-units-fcl fp-utils"
+  licenserecon_repo: "https://git.launchpad.net/ubuntu/+source/licenserecon"
+  licenserecon_branch: "applied/ubuntu/devel"
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+    paths:
+      - "insights/debian/copyright"
+      - "insights/debian/lrc.config"
+      - "insights/go.mod"
+  workflow_dispatch:
 
 jobs:
   run-lrc:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
-      - name: Install dependencies
+      - name: Get latest licenserecon commit hash
+        id: licenserecon-commit
+        run: |
+          if COMMIT_HASH=$(git ls-remote ${{ env.licenserecon_repo }} ${{ env.licenserecon_branch }} | cut -f1); then
+            echo "hash=$COMMIT_HASH" >> $GITHUB_OUTPUT
+            echo "cache-enabled=true" >> $GITHUB_OUTPUT
+            echo "Latest licenserecon commit: $COMMIT_HASH"
+          else
+            echo "::warning::Failed to get latest licenserecon commit hash, proceeding without cache"
+            echo "hash=fallback-$(date +%Y%m%d)" >> $GITHUB_OUTPUT
+            echo "cache-enabled=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Cache licenserecon build - restore
+        uses: actions/cache/restore@v4
+        id: cache-licenserecon-restore
+        if: steps.licenserecon-commit.outputs.cache-enabled == 'true'
+        with:
+          path: licenserecon_*.deb
+          key: ${{ runner.os }}-licenserecon-${{ steps.licenserecon-commit.outputs.hash }}
+          restore-keys: |
+            ${{ runner.os }}-licenserecon-
+
+      - name: Install build dependencies
+        if: steps.licenserecon-commit.outputs.cache-enabled == 'false' || steps.cache-licenserecon-restore.outputs.cache-hit != 'true'
         run: |
           sudo apt update
-          sudo apt install -y ${{ env.apt_deps }}
+          sudo apt install -y ${{ env.lrc_build_deps }}
+
+      - name: Clone and build licenserecon
+        id: build-licenserecon
+        if: steps.licenserecon-commit.outputs.cache-enabled == 'false' || steps.cache-licenserecon-restore.outputs.cache-hit != 'true'
+        run: |
+          git clone -b ${{ env.licenserecon_branch }} ${{ env.licenserecon_repo }}
+          cd licenserecon
+          dpkg-buildpackage -us -uc -b
+
+      - name: Install licenserecon
+        run: |
+          sudo dpkg -i licenserecon_*.deb || sudo apt-get install -f -y
```

Comment:
I was wondering whether the Debian ceremonies were worth in this context or not, specially because building `lrc` can be done with simply `cd .. && make `, but I think in the long run it is indeed better to just relying on the common interface of building a debian package versus dealing with potential complexities of less often used languages and build systems that may be introduced over time. Good move!